### DISCUSSION TITLE
Image dimensions

### DISCUSSION THREAD

**Discussion by mynona** [soto-project/soto#642] [Q&A]
Is it possible to read the image dimensions (width, height) when I use soto to upload files to amazon AWS?

Reason: Deliver width/height attributes to make rendering faster in the browser.

**Comment by adam-fowler** [comment]
I assume you are storing your image in S3. When you upload your images you can include additional metadata that includes the image dimensions. eg
```swift
let request = S3.PutObjectRequest(
    body: .data(imageData),
    bucket: bucketName,
    contentType: "image/jpeg",
    key: filename,
    metadata: ["width": String(describing: imageWidth), "height": String(describing: imageHeight)]
)
try await s3.putObject(request)
```
After that when you call getObject or headObject the metadata field should include the width and height
